### PR TITLE
Basic OAuth support added

### PR DESCRIPTION
**Pull request by attrib**
Addes OAuth support #23

### CODE REVIEW

File: src/jira.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ export default class JiraApi {
   constructor(options) {
     this.protocol = options.protocol || 'http';
     this.host = options.host;
-    this.port = options.port || 80;
+    this.port = options.port || null;
```
Comment (KanoYugoro) [inline src/jira.js]: Why did you change the default port?

File: test/jira-tests.js [hunk 2]
```diff
@@ -40,6 +41,24 @@ describe('Jira API Tests', () => {
       expect(jira.baseOptions.auth).to.be.undefined;
     });
 
+    it('Constructor with oauth credentials', () => {
+      const options = getOptions({ oauth: {
+        consumer_key: 'consumer',
+        consumer_secret: 'consumer_secret',
+        access_token: 'token',
+        access_token_secret: 'token_secret'
+      } });
```
Comment (KanoYugoro) [inline test/jira-tests.js]: This sort of tabbing is slightly inconsistent with the remainder of the file.  Not trying to be nitpicky or difficult, I only bring this up because I nearly missed the `oauth: {}` scope when reading this over.

```javascript
const options = getOptions({ 
  oauth: {
    consumer_key: 'consumer',
    consumer_secret: 'consumer_secret',
    access_token: 'token',
    access_token_secret: 'token_secret'
  } 
});
```

File: test/jira-tests.js [hunk 3]
```diff
@@ -40,6 +41,24 @@ describe('Jira API Tests', () => {
...
+
+      const jira = new JiraApi(options);
+
+      expect(jira.baseOptions.auth).to.be.undefined;
+      expect(jira.baseOptions.oauth.consumer_key).to.eql('consumer');
+      expect(jira.baseOptions.oauth.consumer_secret).to.eql('consumer_secret');
+      expect(jira.baseOptions.oauth.token).to.eql('token');
+      expect(jira.baseOptions.oauth.token_secret).to.eql('token_secret');
+      expect(jira.baseOptions.oauth.signature_method).to.eql('RSA-SHA1');
```
Comment (KanoYugoro) [inline test/jira-tests.js]: In the event that this test doesn't pass, let's say because `jira.baseOptions.oauth.token` is not the expected value of `'token'` and because `jira.baseOptions.oauth.token_secret` is not the expected value of `'token_secret'`, you would have to run this test twice to find that out.  The test will fail on the first `expect` failure, so the first time would break on the expect around token, and then the second time would break around token_secret.

It is preferrable to do your expect statement as follows:
```javascript
expect(jira.baseOptions.oauth).to.eql({
  consumer_key: 'consumer',
  consumer_secret: 'consumer_secret',
  token: 'token',
  token_secret: 'token_secret',
  signature_method: 'RSA-SHA1'
});
```
This way, if `jira.baseOptions.oauth.signature_method` is not the whole object we expect, the testing framework will show us every property that is missing, different, the same, or extra.

File: src/jira.js (hunk 1 above)
Comment (attrib) [inline src/jira.js]: Thanks for the review, will look over the stuff later.
About the default port change. As long as you use port 80 or 443 its just not needed to add this.
http://aaaa.com is the same as http://aaaa.com:80. Adding the 80 is unnecessary and the function to create the url can handle this.

Why I change this is because of oauth. Because at least jira atlassian is creating the string for comparision without the port (e.g. http://xxx.atlassian.net), while when settings this not to null, request is doing it with the port. Then the tokens doesn't match and auth fails. Toke me 2.5h to find this out and I only needed to remove the port. So that nobody falls in the same trap, the port should only be set if its a none standard port.

Comment (mtscout6) [inline src/jira.js]: That sounds reasonable enough to change. We can still push this out as a Major version bump anyways. I've been holding off on #19 since it's also a breaking change so we can bundle the two in the next release.

**Comment by KanoYugoro** [comment]
Hey, it's been a few days.  I would like to pull this in before I do a major version release, but if I can't it will be a while before this gets in.  Any ETA on making those changes?

**Comment by KanoYugoro** [comment]
Hey, I rebased your changes on the latest master and pulled them in.
